compiler/prelude/prelude.go

```diff
@@ -372,7 +372,7 @@ var $internalAppend = function(slice, array, offset, length) {
 
 var $equal = function(a, b, type) {
   if (type === $jsObjectPtr) {
-    return a === b;
+    return a === b || (a.hasOwnProperty("object") && a.object === undefined && b.hasOwnProperty("object") && b.object === undefined);
```

Comment:
Unless I change something else elsewhere, `a === b` did not hold. That's why I needed to add the or block. If you remove it, tests will not pass.